src/plots/smith/layout_defaults.js

```diff
@@ -0,0 +1,151 @@
+
+        coerceAxis('uirevision', contOut.uirevision);
+
+        var dfltColor;
+        var dfltFontColor;
+
+        if(visible) {
+            dfltColor = coerceAxis('color');
+            dfltFontColor = (dfltColor === axIn.color) ? dfltColor : opts.font.color;
+        }
+
+        // We don't want to make downstream code call ax.setScale,
+        // as both radial and angular axes don't have a set domain.
+        // Furthermore, angular axes don't have a set range.
+        //
+        // Mocked domains and ranges are set by the smith subplot instances,
+        // but Axes.findExtremes uses the sign of _m to determine which padding value
+        // to use.
+        //
+        // By setting, _m to 1 here, we make Axes.findExtremes think that
+        // range[1] > range[0], and vice-versa for `autorange: 'reversed'` below.
+        axOut._m = 1;
+
+        switch(axName) {
+            case 'realaxis':
+                axIn.autorange = false;
+                axOut.cleanRange('range', {dfltRange: [0, 1]});
+
+                if(visible) {
+                    coerceAxis('title.text');
+                    Lib.coerceFont(coerceAxis, 'title.font', {
+                        family: opts.font.family,
+                        size: Math.round(opts.font.size * 1.2),
+                        color: dfltFontColor
+                    });
+                }
+                break;
+
+            case 'imaginaryaxis':
+                var direction = coerceAxis('direction');
+                coerceAxis('rotation', {counterclockwise: 0, clockwise: 90}[direction]);
+                break;
+        }
+
+        if(visible) {
+            handleTickMarkDefaults(axIn, axOut, coerceAxis, {outerTicks: true});
+
+            var showTickLabels = coerceAxis('showticklabels');
+            if(showTickLabels) {
+                Lib.coerceFont(coerceAxis, 'tickfont', {
+                    family: opts.font.family,
+                    size: opts.font.size,
+                    color: dfltFontColor
+                });
+                coerceAxis('tickangle');
+                coerceAxis('tickformat');
+            }
+
+            handleLineGridDefaults(axIn, axOut, coerceAxis, {
+                dfltColor: dfltColor,
+                bgColor: opts.bgColor,
+                // default grid color is darker here (60%, vs cartesian default ~91%)
+                // because the grid is not square so the eye needs heavier cues to follow
+                blend: 60,
+                showLine: true,
+                showGrid: true,
+                noZeroLine: true,
+                attributes: layoutAttributes[axName]
+            });
+
+            coerceAxis('layer');
+        }
+
+        if(axType !== 'category') coerceAxis('hoverformat');
+
+        axOut._input = axIn;
+    }
+
+    if(contOut.imaginaryaxis.type === 'category') {
+        coerce('gridshape');
+    }
+}
+
+function handleAxisTypeDefaults(axIn, axOut) {
+    axOut.type = 'linear';
```

Comment:
Do we have to add *"linear"* `type` to this object?

Reply:
Currently, yes. However, if we are able to simplify the "calc" workflow in this trace, then maybe we can get rid of this. I need some help to understand how to refactor and simplify that workflow.

Reply:
@alexcjohnson what is your suggestion here?